### PR TITLE
Add docs in specs for types

### PR DESCRIPTION
**Pull request by leighmcculloch**
### What
Add docs in specs for types.

### Why
To further expand and fill out what we store docs for in specs.

### CODE REVIEW

**Comment by leighmcculloch** [comment]
Todo:
- [ ] Tests.

File: soroban-sdk/src/tests/contract_docs_enum.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
+use crate as soroban_sdk;
```
Comment (dmkozh) [inline soroban-sdk/src/tests/contract_docs_enum.rs:1]: I'm not strictly against creating multiple test files, but just wonder if there is something preventing us from consolidating the tests into a single file.

File: soroban-sdk/src/tests/contract_docs_enum_int.rs [hunk 2]
```diff
@@ -0,0 +1,38 @@
+use crate as soroban_sdk;
+use soroban_sdk::contracttype;
+use stellar_xdr::{ReadXdr, ScSpecEntry, ScSpecUdtEnumCaseV0, ScSpecUdtEnumV0};
+
+/// E has variants A and B.
+#[contracttype]
+#[derive(Copy, Clone)]
+pub enum E {
+    /// A is a.
```
Comment (dmkozh) [inline soroban-sdk/src/tests/contract_docs_enum_int.rs:9]: Can we cover a multi-line comment for enum variant somewhere?

**Review by dmkozh — APPROVED** [review]

File: soroban-sdk/src/tests/contract_docs_enum.rs (hunk 1 above)
Comment (leighmcculloch) [reply soroban-sdk/src/tests/contract_docs_enum.rs:1]: Nothing prevents it. Anything we can do in a file we can do inside a mod inside one file. I'll try that before merging to see if it's too overwhelming.

Comment (leighmcculloch) [reply soroban-sdk/src/tests/contract_docs_enum.rs]: Consolidated the tests in one big file: 3d6c7f63

File: soroban-sdk/src/tests/contract_docs_enum_int.rs (hunk 2 above)
Comment (leighmcculloch) [reply soroban-sdk/src/tests/contract_docs_enum_int.rs]: Added some multiline cases in fce2c958.

**Pull request merged**